Ruby style cleanups

anttijk opened the pull request:
#### 4be388cadb468d7b951a79e5ea6940a3aec607e6
<pre>
Ruby style cleanups
<a href="https://bugs.webkit.org/show_bug.cgi?id=266623">https://bugs.webkit.org/show_bug.cgi?id=266623</a>
<a href="https://rdar.apple.com/119856171">rdar://119856171</a>

Reviewed by Tim Nguyen and Alan Baradlay.

* LayoutTests/fast/ruby/ruby-block-style-not-updated-expected.txt:
* Source/WebCore/html/RubyElement.cpp:
(WebCore::RubyElement::createElementRenderer):
* Source/WebCore/html/RubyTextElement.cpp:
(WebCore::RubyTextElement::createElementRenderer):

Ensure we can&apos;t create legacy ruby renderers with css ruby enabled.

* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::forceBidiIsolationForRuby):

Rename and include all cases to switch.

(WebCore::Style::Adjuster::adjust const):

Remove some no-op code.

(WebCore::Style::adjustUnicodeBidiForRuby): Deleted.

Canonical link: <a href="https://commits.webkit.org/272267@main">https://commits.webkit.org/272267@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14992ae0c864c37f3c15935d5496cce09f51ad78

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31149 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9824 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32840 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33655 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/28283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31915 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12167 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7082 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27948 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31484 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8281 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27847 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7113 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7287 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27742 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34997 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28348 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28199 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33422 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7323 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5379 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31261 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9009 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7328 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8026 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7857 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/style/StyleAdjuster.cpp:
```diff
@@ -379,17 +379,21 @@ static bool isRubyContainerOrInternalRubyBox(const RenderStyle& style)
 {
     switch (unicodeBidi) {
     case UnicodeBidi::Normal:
     case UnicodeBidi::Embed:
+    case UnicodeBidi::Isolate:
         return UnicodeBidi::Isolate;
     case UnicodeBidi::Override:
+    case UnicodeBidi::IsolateOverride:
         return UnicodeBidi::IsolateOverride;
-    default:
-        return unicodeBidi;
+    case UnicodeBidi::Plaintext:
+        return UnicodeBidi::Plaintext;
     }
+    ASSERT_NOT_REACHED();
+    return UnicodeBidi::Isolate;
```
I actually think the old code read more similarly to the spec. No strong opinion here though.

> Bidi isolation is forced on all [internal ruby boxes](https://drafts.csswg.org/css-ruby-1/#internal-ruby-boxes) and the [ruby container](https://drafts.csswg.org/css-ruby-1/#ruby-container): the [normal](https://drafts.csswg.org/css-writing-modes-4/#valdef-unicode-bidi-normal) and [embed](https://drafts.csswg.org/css-writing-modes-4/#valdef-unicode-bidi-embed) values of [unicode-bidi](https://drafts.csswg.org/css-writing-modes-3/#propdef-unicode-bidi) compute to [isolate](https://drafts.csswg.org/css-writing-modes-4/#valdef-unicode-bidi-isolate), and [bidi-override](https://drafts.csswg.org/css-writing-modes-4/#valdef-unicode-bidi-bidi-override) computes to [isolate-override](https://drafts.csswg.org/css-writing-modes-4/#valdef-unicode-bidi-isolate-override).

anttijk committed:
Ruby style cleanups
https://bugs.webkit.org/show_bug.cgi?id=266623
rdar://119856171

Reviewed by Tim Nguyen and Alan Baradlay.

* LayoutTests/fast/ruby/ruby-block-style-not-updated-expected.txt:
* Source/WebCore/html/RubyElement.cpp:
(WebCore::RubyElement::createElementRenderer):
* Source/WebCore/html/RubyTextElement.cpp:
(WebCore::RubyTextElement::createElementRenderer):

Ensure we can't create legacy ruby renderers with css ruby enabled.

* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::forceBidiIsolationForRuby):

Rename and include all cases to switch.

(WebCore::Style::Adjuster::adjust const):

Remove some no-op code.

(WebCore::Style::adjustUnicodeBidiForRuby): Deleted.

Canonical link: https://commits.webkit.org/272267@main
